docs/advanced/authentication.md

```diff
@@ -0,0 +1,328 @@
+    return operation;
+  }
+
+  const fetchOptions =
+    typeof operation.context.fetchOptions === 'function'
+      ? operation.context.fetchOptions()
+      : operation.context.fetchOptions || {};
+
+  return {
+    ...operation,
+    context: {
+      ...operation.context,
+      fetchOptions: {
+        ...fetchOptions,
+        headers: {
+          ...fetchOptions.headers,
+          "Authorization": authState.token,
+        },
+      },
+    },
+  };
+}
+```
+
+First we check that we have an `authState` and a `token`. Then we apply it to the request `fetchOptions` as an `Authorization` header.
+The header format can vary based on the API (e.g using `Bearer ${token}` instead of just `token`) which is why it'll be up to you to add the header
+in the expected format for your API.
+
+### Configuring `didAuthError`
+
+This function lets the exchange know what is defined to be an API error for your API:
+
+```js
+const didAuthError = ({ error }) => {
+  return error.graphQLErrors.some(
+    e => e.extensions?.code === 'FORBIDDEN',
+  );
+}
+```
+
+For most graphQL APIs, the auth error is communicated as an error code, however it may also be a 401 HTTP response:
+
+```js
+const didAuthError = ({ error }) => {
+  return error.graphQLErrors.some(
+    e => error.response.status === 401,
+  );
+},
+```
+
+Then `didAuthError` returns `true`, it will trigger the exchange to trigger the logic for asking for reauthentication via `getAuth`.
+
+### Configuring `getAuth` (after `didAuthError` returns `true`)
+
+If your API doesn't support any sort of token refresh, this is where you should simply log the user out.
+
+```js
+const getAuth = async ({ authState }) => {
+  if (!authState) {
+    const token = localStorage.getItem('token');
+    const refreshToken = localStorage.getItem('refreshToken');
+    if (token && refreshToken) {
+      return { token, refreshToken };
+    }
+    return null;
+  }
+
+  logout();
+
+  return null;
+}
+```
+
+Here, `logout()` should clear your persisted storage of any tokens and set the `isLoggedIn` property as described above to `false`.
+
+If you do have a token refresh endpoint, and a refresh token, you can attempt to get a new token for the user:
+
+```js
+const getAuth = async ({ authState, mutate }) => {
+  if (!authState) {
+    const token = localStorage.getItem('token');
+    const refreshToken = localStorage.getItem('refreshToken');
+    if (token && refreshToken) {
+      return { token, refreshToken };
+    }
+    return null;
+  }
+
+  const result = await mutate(refreshMutation, {
+    token: authState!.refreshToken,
+  });
+
+  if (result.data?.refreshLogin) {
+    localStorage.setItem('token', result.data.refreshLogin.token);
+    localStorage.setItem('refreshToken', result.data.refreshLogin.refreshToken);
+
+    return {
+      token: result.data.refreshLogin.token,
+      refreshToken: result.data.refreshLogin.refreshToken,
+    };
+  }
+
+  localStorage.clear();
+  logout();
```

Comment:
```suggestion
  // This is where auth has gone wrong and we need to clean up and redirect to a login page
  localStorage.clear();
  logout();
```

Just adding a comment here since this isn't reusable for the user's app